app/test/vesting_test.go

```diff
@@ -0,0 +1,550 @@
+
+	msg := distributiontypes.NewMsgWithdrawDelegatorReward(
+		getAddress(accName, cctx.Keyring),
+		validators[0].GetOperator(),
+	)
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	resTx, err := testnode.SignAndBroadcastTx(encCfg, cctx.Context, accName, []sdk.Msg{msg}...)
+	assert.NoError(s.T(), err)
+
+	resQ, err := cctx.WaitForTx(resTx.TxHash, 10)
+	assert.NoError(s.T(), err)
+	assert.EqualValues(s.T(), 0, resQ.TxResult.Code, "the claim reward TX must succeed")
+
+	// Check if the reward amount in the account
+	balancesAfter, err := GetAccountSpendableBalance(cctx.GRPCClient, accAddress)
+	assert.NoError(s.T(), err)
+
+	assert.GreaterOrEqual(s.T(), balancesAfter.AmountOf(app.BondDenom).Int64(), minExpectedBalance, "Minimum balance after claiming reward")
+}
+
+func AddAccountsToGenesisState(gs map[string]json.RawMessage, accounts ...authtypes.GenesisAccount) (map[string]json.RawMessage, error) {
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	var authGenState authtypes.GenesisState
+	err := encCfg.Codec.UnmarshalJSON(gs[authtypes.ModuleName], &authGenState)
+	if err != nil {
+		return gs, err
+	}
+
+	pAccs, err := authtypes.PackAccounts(accounts)
+	if err != nil {
+		return gs, err
+	}
+
+	// set the accounts in the genesis state
+	authGenState.Accounts = append(authGenState.Accounts, pAccs...)
+	gs[authtypes.ModuleName] = encCfg.Codec.MustMarshalJSON(&authGenState)
+
+	return gs, nil
+}
+
+func AddBalancesToGenesisState(gs map[string]json.RawMessage, balances ...banktypes.Balance) (map[string]json.RawMessage, error) {
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	var bankGenState banktypes.GenesisState
+	err := encCfg.Codec.UnmarshalJSON(gs[banktypes.ModuleName], &bankGenState)
+	if err != nil {
+		return gs, err
+	}
+
+	bankGenState.Balances = append(bankGenState.Balances, balances...)
+	gs[banktypes.ModuleName] = encCfg.Codec.MustMarshalJSON(&bankGenState)
+
+	return gs, nil
+}
+
+func GetValidators(grpcConn *grpc.ClientConn) (stakingtypes.Validators, error) {
+	scli := stakingtypes.NewQueryClient(grpcConn)
+	vres, err := scli.Validators(context.Background(), &stakingtypes.QueryValidatorsRequest{})
+
+	if vres == nil {
+		return stakingtypes.Validators{}, err
+	}
+	return vres.Validators, err
+}
+
+func GetAccountDelegations(grpcConn *grpc.ClientConn, address string) (stakingtypes.DelegationResponses, error) {
+	cli := stakingtypes.NewQueryClient(grpcConn)
+	res, err := cli.DelegatorDelegations(context.Background(),
+		&stakingtypes.QueryDelegatorDelegationsRequest{DelegatorAddr: address})
+	if err != nil {
+		return nil, err
+	}
+
+	return res.DelegationResponses, err
+}
+
+func GetAccountSpendableBalance(grpcConn *grpc.ClientConn, address string) (sdk.Coins, error) {
+	cli := banktypes.NewQueryClient(grpcConn)
+	res, err := cli.SpendableBalances(
+		context.Background(),
+		&banktypes.QuerySpendableBalancesRequest{
+			Address: address,
+		},
+	)
+	if err != nil || res == nil {
```

Comment:
[question] is the second part of this conditional necessary? If not, proposal to remove so that this follows the expected pattern

```suggestion
	if err != nil {
```

Reply:
it is to avoid panic on `res.GetBalances()` if `res` is nil. So if the address is not found, `res` will be nil.